### ISSUE TITLE
Incorrect `Error: It's not possible to meet the minimal approval count of 1`

### ISSUE THREAD

**Issue by koute** [paritytech/parity-processbot#347]
I tried to merge this PR:

https://github.com/paritytech/substrate/pull/10149

while I had 1 approval (from bkchr) in my companion:

https://github.com/paritytech/polkadot/pull/4210

and yet the bot gave me this error:

```
Error: It's not possible to meet the minimal approval count of 1 in paritytech/polkadot#4210
```

Only after @ordian added his own approval to https://github.com/paritytech/polkadot/pull/4210 did the "bot merge" work in https://github.com/paritytech/substrate/pull/10149.

After this [the Polkadot companion merge failed anyway](https://github.com/paritytech/polkadot/pull/4210#issuecomment-964093304) which may or may not be related. (Although I'd argue it shouldn't fail; isn't the whole point of the bot to make the merge across two repositories atomic?)

**Comment by joao-paulo-parity** [comment]
> After this the Polkadot companion merge failed anyway which may or may not be related. (Although I'd argue it shouldn't fail; isn't the whole point of the bot to make the merge across two repositories atomic?)

The current design inherently is flawed in this regard. For the merge to be "atomic" we'd need:

1. Full CI integration (https://github.com/paritytech/ci_cd/issues/234#issue-1026474765) so that all PRs are green and therefore the API merge call can be fired simultaneously on all of them at once. Currently we first merge one PR while their companions are red under the pretense of "after updating the reference, the companion will become green" but obviously there's no guarantee that this is actually the case in practice - for instance, we've seen the companion fail due to missing labels as reported in https://github.com/paritytech/ci_cd/issues/159#issuecomment-899549998.
2. A queue which prioritizes companions, which was [brought up before](https://github.com/paritytech/parity-processbot/issues?q=is%3Aissue+is%3Aopen+queue)

It is **necessary** to redesign the companion build system - https://github.com/paritytech/parity-processbot/issues/327 and many other stray comments in this issue tracker hint at that - or possibly move it out of the CI pipeline so that we'll have less friction for developers and external contributors alike. I have been talking with @TriplEight about those ideas precisely. Last year I did not have time to work on this - had too many projects + I failed to raise exposure for this problem - but it's something that should be dealt with.

**Comment by joao-paulo-parity** [comment]
The "It's not possible to meet the minimal approval count of 1" error should be fixed https://github.com/paritytech/parity-processbot/pull/358/files#diff-ee991933fcc804e8dec3619740cb135ffc5a3d19bf55336acecd8b5f05849032R1180. I'll re-open or file a new issue in case it happens again.

The Polkadot update failure is being tracked on https://github.com/paritytech/parity-processbot/issues/348.

Closing as there's no action left to take here.
